azavea/tilejson.io
# Improve the share experience and add functionality

## Overview

Share screen now has a smaller navbar.
The share page also allows diffs. This removes the need for having separate diff and regular shares.
Share options available prior to share. These allow the user to include the base layer, a link to tilejson.io, a link to the github gist, and the option to allow diffs on the shared map. If the diff mode is enabled, will allow the user to default the shared map to diff mode. 
This also fixes an error thrown when "copy link" is clicked. 

Connects #49 

### Demo

<img width="1280" alt="screen shot 2017-11-24 at 12 02 16 pm" src="https://user-images.githubusercontent.com/3959096/33219460-4bca7112-d110-11e7-996f-8ca0fafb0561.png">

<img width="1280" alt="screen shot 2017-11-24 at 12 04 38 pm" src="https://user-images.githubusercontent.com/3959096/33219464-4e822710-d110-11e7-8f55-8c8b4607295f.png">

<img width="1280" alt="screen shot 2017-11-24 at 12 05 43 pm" src="https://user-images.githubusercontent.com/3959096/33219465-516a5c4a-d110-11e7-91dd-cf8053f1ae69.png">

<img width="1280" alt="screen shot 2017-11-24 at 12 08 07 pm" src="https://user-images.githubusercontent.com/3959096/33219469-573654f8-d110-11e7-8732-7865d92f917e.png">

<img width="1280" alt="screen shot 2017-11-24 at 12 08 13 pm" src="https://user-images.githubusercontent.com/3959096/33219475-5ac5e12e-d110-11e7-8a3c-3a37d1cad7c2.png">


## Testing Instructions

 * Add one layer. Share with no title, no description, and all the options unchecked. See that you get just a map with the one layer (no base layer), no card, no navbar. 
 * Now share with a title and/or a description, and all the options still unchecked. See that you get just a map with one layer and a card with title and/or description.
 * Now share with the base layer option only and see that the base layer is visible. 
 * Share with the the links only, see that the links you shared are visible. 
 * Share with combinations of the above and see if the shared map is as desired.
 * Add a second (or more) layers and share with the diff, without the diff, with defaulting to diff.
 * Enter diff mode on the app and share. See that the diff options default to true.
